File: src/lib/local-api/handlers/knowledge.ts
/**
 * 知识模块 handler
 * GET /api/v1/knowledge/categories
 * GET /api/v1/knowledge/articles?category=...
 * GET /api/v1/knowledge/articles/{id}
 */

import { getLocalDB } from '../../local-db'
import type { LocalRequest, LocalResponse } from '../router'

function ok(data: any): LocalResponse {
  return { error: false, message: 'ok', data }
}

export async function handleKnowledge(req: LocalRequest, route: string): Promise<LocalResponse> {
  const db = await getLocalDB()

  if (route === 'knowledge/categories') {
    const r = await db.query(`
      SELECT id, name, slug, icon, sort_order
      FROM jx_apk_knowledge_categories
      WHERE status = 1
      ORDER BY sort_order ASC
    `)
    return ok({ items: r.rows })
  }

  if (route === 'knowledge/articles') {
    const { category, page = '1', page_size = '20' } = req.query
    const offset = (Number(page) - 1) * Number(page_size)
    const r = await db.query(
      `
      SELECT a.id, a.article_uuid, a.title, a.summary, a.cover_image,
             a.category_id, a.is_video, a.video_url, a.view_count,
             a.share_count, a.favorite_count, a.published_at,
             c.name as category_name
      FROM jx_apk_knowledge_articles a
      LEFT JOIN jx_apk_knowledge_categories c ON c.id = a.category_id
      WHERE a.status = 1
        AND ($1::text IS NULL OR a.category_id = $1)
      ORDER BY a.is_featured DESC, a.sort_order ASC
      LIMIT $2 OFFSET $3
    `,
      [category && category !== 'all' ? category : null, Number(page_size), offset]
    )
    const totalR = await db.query(
      `
      SELECT count(*)::int as total FROM jx_apk_knowledge_articles
      WHERE status = 1 AND ($1::text IS NULL OR category_id = $1)
    `,
      [category && category !== 'all' ? category : null]
    )
    const total: number = (totalR.rows[0] as any).total
    return ok({
      items: r.rows,
      total,
      page: Number(page),
      page_size: Number(page_size),
      total_pages: Math.ceil(total / Number(page_size)),
    })
  }

  const detailMatch = route.match(/^knowledge\/articles\/(.+)$/)
  if (detailMatch) {
    const idOrUuid = detailMatch[1]
    const r = await db.query(
      `
      SELECT a.*, c.name as category_name
      FROM jx_apk_knowledge_articles a
      LEFT JOIN jx_apk_knowledge_categories c ON c.id = a.category_id
      WHERE a.id::text = $1 OR a.article_uuid = $1
      LIMIT 1
    `,
      [idOrUuid]
    )
    if (r.rows.length === 0) return { error: true, message: '文章不存在', data: null }
    return ok(r.rows[0])
  }

  return { error: true, message: `Unknown knowledge route: ${route}`, data: null }
}

File: backend/app/models/knowledge.py
import uuid
from datetime import datetime
from sqlalchemy import String, Text, Integer, Boolean, ForeignKey, DateTime, Uuid
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.models.base import Base, TimestampMixin


class KnowledgeCategory(Base, TimestampMixin):
    """
    知识分类表
    """
    __tablename__ = "jx_apk_knowledge_categories"

    id: Mapped[str] = mapped_column(String(100), primary_key=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    slug: Mapped[str] = mapped_column(String(200), unique=True, nullable=False)
    icon: Mapped[str | None] = mapped_column(String(100), nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    status: Mapped[int] = mapped_column(Integer, default=1)
    created_by: Mapped[uuid.UUID | None] = mapped_column(Uuid, nullable=True)

    # 关系
    articles: Mapped[list["KnowledgeArticle"]] = relationship(
        "KnowledgeArticle",
        back_populates="category",
        cascade="all, delete-orphan"
    )


class KnowledgeArticle(Base, TimestampMixin):
    """
    知识文章表
    包含图文和视频类型文章
    """
    __tablename__ = "jx_apk_knowledge_articles"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    article_uuid: Mapped[uuid.UUID] = mapped_column('uuid', Uuid, unique=True, default=uuid.uuid4)

    # 基础信息
    title: Mapped[str] = mapped_column(String(500), nullable=False)
    summary: Mapped[str] = mapped_column(Text, nullable=True)
    content: Mapped[str] = mapped_column(Text, nullable=True)  # 富文本内容

    # 分类
    category_id: Mapped[str | None] = mapped_column(
        String(100),
        ForeignKey("jx_apk_knowledge_categories.id", ondelete="SET NULL"),
        nullable=True
    )

    # 封面和媒体
    cover_image: Mapped[str | None] = mapped_column(String(500), nullable=True)
    is_video: Mapped[bool] = mapped_column(Boolean, default=False)
    video_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    video_duration: Mapped[int | None] = mapped_column(Integer, nullable=True)  # 视频时长(秒)

    # 统计
    view_count: Mapped[int] = mapped_column(Integer, default=0)
    share_count: Mapped[int] = mapped_column(Integer, default=0)
    favorite_count: Mapped[int] = mapped_column(Integer, default=0)

    # 状态
    status: Mapped[int] = mapped_column(Integer, default=1)  # 0:下架 1:上架
    is_featured: Mapped[bool] = mapped_column(Boolean, default=False)  # 是否精选
    sort_order: Mapped[int] = mapped_column(Integer, default=0)

    # 发布信息
    published_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_by: Mapped[uuid.UUID | None] = mapped_column(Uuid, nullable=True)

    # 关系
    category: Mapped["KnowledgeCategory | None"] = relationship(
        "KnowledgeCategory",
        back_populates="articles"
    )